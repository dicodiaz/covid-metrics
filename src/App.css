@import url('https://fonts.googleapis.com/css2?family=Lato:ital,wght@0,100;0,300;0,400;0,700;0,900;1,100;1,300;1,400;1,700;1,900&display=swap');

body {
  font-family: gillsans, sans-serif;
}

.font-lato {
  font-family: lato, sans-serif !important;
}

.text-custom1 {
  color: #4369b2 !important;
}

.bg-custom1 {
  background-color: #4369b2 !important;
}
