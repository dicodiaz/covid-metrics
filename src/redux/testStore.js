import { applyMiddleware, combineReducers, createStore } from 'redux';
import thunk from 'redux-thunk';
import covidReducer from './ducks/covid';

const reducer = combineReducers({
  covidReducer,
});

const store = createStore(reducer, applyMiddleware(thunk));

export default store;
