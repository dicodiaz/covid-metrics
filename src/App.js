import React from 'react';
import { Navigate, Route, Routes } from 'react-router-dom';
import './App.css';
import Details from './components/Details';
import Home from './components/Home';

const App = () => (
  <>
    <Routes>
      <Route path="/" element={<Home />} />
      <Route path="/details/:country" element={<Details />} />
      <Route path="*" element={<Navigate to="/" />} />
    </Routes>
  </>
);

export default App;
