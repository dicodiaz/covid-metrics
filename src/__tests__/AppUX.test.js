import { fireEvent, render, within } from '@testing-library/react';
import fetchMock from 'fetch-mock';
import { Provider } from 'react-redux';
import App from '../App';
import testStore from '../redux/testStore';
import getMockCovidData from '../__mocks__/getMockCovidData';

global.scrollTo = jest.fn();
const delay = (ms) => new Promise((resolve) => setTimeout(resolve, ms));

let tree;
describe('testing UX for the whole App', () => {
  const url = 'https://api.covid19tracking.narrativa.com/api/';
  fetchMock.get(url, getMockCovidData());

  beforeEach(() => {
    fetchMock.reset();
    tree = render(
      <Provider store={testStore}>
        <App />
      </Provider>,
    );
  });

  test('testing homepage, details page, navigation between both, and snapshot', async () => {
    // homepage
    await delay(1000);
    const countries = await tree.findAllByTestId('country');
    expect(countries).toHaveLength(195);
    const inputCountry = await tree.findByRole('textbox');
    fireEvent.change(inputCountry, { target: { value: 'Colombia' } });
    const country = await tree.findByTestId('country');
    expect(await within(country).findByText('Colombia')).toBeTruthy();
    expect(await tree.queryByTestId('backBtn')).toBeFalsy();
    // navigation to details page
    fireEvent.click(country);
    // details page
    const regions = await tree.findAllByTestId('region');
    expect(regions).toHaveLength(33);
    const inputRegion = await tree.findByRole('textbox');
    fireEvent.change(inputRegion, { target: { value: 'Bogotá' } });
    const region = await tree.findByTestId('region');
    expect(await within(region).findByText('Bogotá')).toBeTruthy();
    expect(await tree.queryByTestId('backBtn')).toBeTruthy();
    // navigation to homepage
    const backBtn = await tree.findByTestId('backBtn');
    fireEvent.click(backBtn);
    expect(tree.queryAllByTestId('country')).toBeTruthy();
    // snapshot
    expect(tree).toMatchSnapshot();
  });
});
