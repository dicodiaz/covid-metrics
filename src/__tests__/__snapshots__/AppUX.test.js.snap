// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing UX for the whole App testing homepage, details page, navigation between both, and snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <header
        class="py-2 bg-custom2 text-white fixed-top"
      >
        <div
          class="container-md"
        >
          <div
            class="mx-0 g-0 justify-content-center align-items-center row"
          >
            <div
              class="col-1"
              data-testid="backBtn"
            >
              <svg
                class="fs-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
                />
              </svg>
            </div>
            <div
              class="col-md-11 col-9"
            >
              <h1
                class="mb-0 text-center"
              >
                Colombia
              </h1>
            </div>
            <div
              class="px-1 d-md-none col-1"
            >
              <svg
                class="fs-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
                />
              </svg>
            </div>
            <div
              class="px-1 d-md-none col-1"
            >
              <svg
                class="fs-1"
                fill="currentColor"
                height="1em"
                stroke="currentColor"
                stroke-width="0"
                viewBox="0 0 24 24"
                width="1em"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0V0z"
                  fill="none"
                />
                <path
                  d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46a.5.5 0 00-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65A.488.488 0 0014 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1a.566.566 0 00-.18-.03c-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46a.5.5 0 00.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
                />
              </svg>
            </div>
          </div>
        </div>
      </header>
      <main
        class="pt-5 bg-custom1 text-white min-vh-100"
      >
        <div
          class="pt-1 pt-sm-2 pt-md-3 pt-lg-4 px-0 container-md"
        >
          <div
            class="mx-0 gy-1 mb-2 row row-cols-1"
          >
            <div
              class="col"
            >
              Total confirmed: 
              5038544
            </div>
            <div
              class="bg-custom3 col"
            >
              Total deaths: 
              127912
            </div>
            <div
              class="col"
            >
              Total recovered: 
              4624669
            </div>
            <div
              class="bg-custom3 col"
            >
              Today's new confirmed: 
              0
            </div>
            <div
              class="col"
            >
              Today's new deaths: 
              0
            </div>
            <div
              class="bg-custom3 col"
            >
              Today's new recovered: 
              0
            </div>
          </div>
          <div
            class="mx-0 gx-0 px-2 row"
          >
            <div
              class="col"
            >
              <input
                class="form-control"
                placeholder="Search..."
                value="Bogotá"
              />
            </div>
            <div
              class="col-auto"
            >
              <button
                class="btn btn-danger"
                type="button"
              >
                Clear
              </button>
            </div>
          </div>
          <h2
            class="text-center h6 my-2"
          >
            STATS BY REGION
          </h2>
          <div
            class="mx-0 gx-0 row row-cols-1"
          >
            <div
              class="col"
            >
              <div
                class="rounded-0 h-100 bg-custom2 card"
                data-testid="region"
              >
                <div
                  class="d-flex flex-column justify-content-center card-body"
                >
                  <div
                    class="row row-cols-2"
                  >
                    <div
                      class="mb-0 d-flex flex-column justify-content-center card-title h5"
                    >
                      Bogotá
                    </div>
                    <div
                      class="col"
                    >
                      <div
                        class="row row-cols-1"
                      >
                        <div
                          class="col"
                        >
                          Confirmed: 
                          1459032
                        </div>
                        <div
                          class="col"
                        >
                          Deaths: 
                          27590
                        </div>
                        <div
                          class="col"
                        >
                          Recovered: 
                          1369593
                        </div>
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </main>
    </div>
  </body>,
  "container": <div>
    <header
      class="py-2 bg-custom2 text-white fixed-top"
    >
      <div
        class="container-md"
      >
        <div
          class="mx-0 g-0 justify-content-center align-items-center row"
        >
          <div
            class="col-1"
            data-testid="backBtn"
          >
            <svg
              class="fs-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M15.41 16.59L10.83 12l4.58-4.59L14 6l-6 6 6 6 1.41-1.41z"
              />
            </svg>
          </div>
          <div
            class="col-md-11 col-9"
          >
            <h1
              class="mb-0 text-center"
            >
              Colombia
            </h1>
          </div>
          <div
            class="px-1 d-md-none col-1"
          >
            <svg
              class="fs-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0z"
                fill="none"
              />
              <path
                d="M12 14c1.66 0 2.99-1.34 2.99-3L15 5c0-1.66-1.34-3-3-3S9 3.34 9 5v6c0 1.66 1.34 3 3 3zm5.3-3c0 3-2.54 5.1-5.3 5.1S6.7 14 6.7 11H5c0 3.41 2.72 6.23 6 6.72V21h2v-3.28c3.28-.48 6-3.3 6-6.72h-1.7z"
              />
            </svg>
          </div>
          <div
            class="px-1 d-md-none col-1"
          >
            <svg
              class="fs-1"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              stroke-width="0"
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M0 0h24v24H0V0z"
                fill="none"
              />
              <path
                d="M19.43 12.98c.04-.32.07-.64.07-.98 0-.34-.03-.66-.07-.98l2.11-1.65c.19-.15.24-.42.12-.64l-2-3.46a.5.5 0 00-.61-.22l-2.49 1c-.52-.4-1.08-.73-1.69-.98l-.38-2.65A.488.488 0 0014 2h-4c-.25 0-.46.18-.49.42l-.38 2.65c-.61.25-1.17.59-1.69.98l-2.49-1a.566.566 0 00-.18-.03c-.17 0-.34.09-.43.25l-2 3.46c-.13.22-.07.49.12.64l2.11 1.65c-.04.32-.07.65-.07.98 0 .33.03.66.07.98l-2.11 1.65c-.19.15-.24.42-.12.64l2 3.46a.5.5 0 00.61.22l2.49-1c.52.4 1.08.73 1.69.98l.38 2.65c.03.24.24.42.49.42h4c.25 0 .46-.18.49-.42l.38-2.65c.61-.25 1.17-.59 1.69-.98l2.49 1c.06.02.12.03.18.03.17 0 .34-.09.43-.25l2-3.46c.12-.22.07-.49-.12-.64l-2.11-1.65zm-1.98-1.71c.04.31.05.52.05.73 0 .21-.02.43-.05.73l-.14 1.13.89.7 1.08.84-.7 1.21-1.27-.51-1.04-.42-.9.68c-.43.32-.84.56-1.25.73l-1.06.43-.16 1.13-.2 1.35h-1.4l-.19-1.35-.16-1.13-1.06-.43c-.43-.18-.83-.41-1.23-.71l-.91-.7-1.06.43-1.27.51-.7-1.21 1.08-.84.89-.7-.14-1.13c-.03-.31-.05-.54-.05-.74s.02-.43.05-.73l.14-1.13-.89-.7-1.08-.84.7-1.21 1.27.51 1.04.42.9-.68c.43-.32.84-.56 1.25-.73l1.06-.43.16-1.13.2-1.35h1.39l.19 1.35.16 1.13 1.06.43c.43.18.83.41 1.23.71l.91.7 1.06-.43 1.27-.51.7 1.21-1.07.85-.89.7.14 1.13zM12 8c-2.21 0-4 1.79-4 4s1.79 4 4 4 4-1.79 4-4-1.79-4-4-4zm0 6c-1.1 0-2-.9-2-2s.9-2 2-2 2 .9 2 2-.9 2-2 2z"
              />
            </svg>
          </div>
        </div>
      </div>
    </header>
    <main
      class="pt-5 bg-custom1 text-white min-vh-100"
    >
      <div
        class="pt-1 pt-sm-2 pt-md-3 pt-lg-4 px-0 container-md"
      >
        <div
          class="mx-0 gy-1 mb-2 row row-cols-1"
        >
          <div
            class="col"
          >
            Total confirmed: 
            5038544
          </div>
          <div
            class="bg-custom3 col"
          >
            Total deaths: 
            127912
          </div>
          <div
            class="col"
          >
            Total recovered: 
            4624669
          </div>
          <div
            class="bg-custom3 col"
          >
            Today's new confirmed: 
            0
          </div>
          <div
            class="col"
          >
            Today's new deaths: 
            0
          </div>
          <div
            class="bg-custom3 col"
          >
            Today's new recovered: 
            0
          </div>
        </div>
        <div
          class="mx-0 gx-0 px-2 row"
        >
          <div
            class="col"
          >
            <input
              class="form-control"
              placeholder="Search..."
              value="Bogotá"
            />
          </div>
          <div
            class="col-auto"
          >
            <button
              class="btn btn-danger"
              type="button"
            >
              Clear
            </button>
          </div>
        </div>
        <h2
          class="text-center h6 my-2"
        >
          STATS BY REGION
        </h2>
        <div
          class="mx-0 gx-0 row row-cols-1"
        >
          <div
            class="col"
          >
            <div
              class="rounded-0 h-100 bg-custom2 card"
              data-testid="region"
            >
              <div
                class="d-flex flex-column justify-content-center card-body"
              >
                <div
                  class="row row-cols-2"
                >
                  <div
                    class="mb-0 d-flex flex-column justify-content-center card-title h5"
                  >
                    Bogotá
                  </div>
                  <div
                    class="col"
                  >
                    <div
                      class="row row-cols-1"
                    >
                      <div
                        class="col"
                      >
                        Confirmed: 
                        1459032
                      </div>
                      <div
                        class="col"
                      >
                        Deaths: 
                        27590
                      </div>
                      <div
                        class="col"
                      >
                        Recovered: 
                        1369593
                      </div>
                    </div>
                  </div>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </main>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
